### PR TITLE
!feat: bl-15046 add static analysis, PHP 8.2

### PR DESCRIPTION
**Pull request by waringr**
## Description

Adds the CI workflow to the repo and adds a phpcs pre-commit hook

Related issue: https://dvsa.atlassian.net/browse/BL-15046

## Before submitting (or marking as "ready for review")

- [x] Does the pull request title follow the [conventional commit](https://www.conventionalcommits.org/en/v1.0.0/) specification?
- [x] Have you performed a self-review of the code?
- [x] Have you have added tests that prove the fix or feature is effective and working?
- [x] Did you make sure to update any documentation relating to this change?

### CODE REVIEW

File: test/Exception/FeatureNotAvailableExceptionTest.php [hunk 1]
```diff
@@ -12,21 +13,30 @@
 
 class FeatureNotAvailableExceptionTest extends TestCase
 {
-    const FEATURE_NAME = 'featureName';
+    private const FEATURE_NAME = 'featureName';
 
+    /**
+     * @return void
```
Comment (faggc) [inline test/Exception/FeatureNotAvailableExceptionTest.php]: Is there a reason the void return type is in a comment and not written as:
`public function testPassingAnEmptyMessageTriggersMessageBuilding(): void`?

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -14,16 +14,18 @@ concurrency:
 
 jobs:
   static:
-    uses: dvsa/.github/.github/workflows/php-library-static.yml@v3.2.3
+    uses: dvsa/.github/.github/workflows/php-library-static.yml@v4.0.1
```
Comment (aaronlong) [inline .github/workflows/ci.yml]: I'm thinking for the CI tasks do we want to run the workflows not for 'library'.  And run these all in prefer-stable mode?

File: test/Exception/FeatureNotAvailableExceptionTest.php (hunk 1 above)
Comment (waringr) [reply test/Exception/FeatureNotAvailableExceptionTest.php]: I think I copied it from somewhere, but it is neater with the actual code syntax, so I've swapped them all over. I'll do it in the other repos as well.

File: .github/workflows/ci.yml (hunk 2 above)
Comment (waringr) [reply .github/workflows/ci.yml]: tbh, I don't know the answer to the first question. I'm cool to go with prefer-stable, but I can't remember where it's set. I saw the `matrix` config you shared yesterday, but I don't know where it is.

**Review by faggc — APPROVED** [review]

File: .github/workflows/ci.yml (hunk 2 above)
Comment (waringr) [reply .github/workflows/ci.yml]: Changed to the non-library versions. I've also bumped the security workflow after it failed at v3.2.3. Unfortunately 4.0.1 also seems to be failing, so I'm going to roll that back now.

Comment (alexisc-kainos) [reply .github/workflows/ci.yml]: What is the error with the latest version of the security action, we just recently released a version of dvsa github v4.1.0 (added a composer-version parameter to php libraries), does this work?

Comment (waringr) [reply .github/workflows/ci.yml]: It's the `Code scanning results / Snyk Open Source` check below, it just says "Snyk Open Source
CI on: pull_request
4
GitHub Advanced Security /[ Snyk Open Source ](https://github.com/dvsa/mot-feature-toggle/pull/5/checks?check_run_id=26506559988)completed Jun 21, 2024 in 4s
1 configuration not found

I'll bump to 4.1.0, but I don't _think_ it's going to change anything.

Comment (waringr) [reply .github/workflows/ci.yml]: It's actually doing the same thing on @alexisc-kainos 's PR, which is on 4.1.0, so I've not bumped anything. I presume that someone has switched something on recently as this wasn't appearing a few days ago.

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -14,16 +14,16 @@ concurrency:
...
     with:
-      php-versions: "[\"8.0\", \"8.1\"]"
+      php-versions: "[\"8.1\"]"
 
   security:
-    uses: dvsa/.github/.github/workflows/php-library-security.yml@v3.2.3
-    with:
-     php-versions: "[\"8.0\", \"8.1\"]"
-    secrets:
+    uses: dvsa/.github/.github/workflows/php-security.yml@v3.2.3
```
Comment (UndermountainK) [inline .github/workflows/ci.yml]: Can we use version @4.1.0 like in this PR: [mot-doctrine-module](https://github.com/dvsa/mot-doctrine-module/pull/10)?

Comment (waringr) [reply .github/workflows/ci.yml]: I've bumped them all to 4.1.0 using php version 8.1 and composer version 2.2.24. My local is running composer 2.7.x, so I figured 2.2.24 would be fine and it seems to be ok.

**Review by alexisc-kainos — APPROVED** [review]

**Review by UndermountainK — APPROVED** [review]

**Review by aaronlong — APPROVED** [review]

**Pull request merged**